http-client/src/test/groovy/io/micronaut/http/client/ClientHostNameSpec.groovy

```diff
@@ -15,7 +15,7 @@ class ClientHostNameSpec extends Specification {
         e.message.contains('Connect Error: foo_bar') || e.message.contains('Connect Error: No such host is known (foo_bar)')
 
         cleanup:
-        client.close()
+        client.refresh()
```

Comment:
why these changes?

Reply:
Cause if close this instance other tests will get an already closed client and then throw an exception when trying to use it

Reply:
the client in this test is not shared

Reply:
When trying to run these tests, I get “closed client” exceptions, when I replace them with refresh, all the tests actually pass. I made quit look inside the impl of factory client method, and it seems to me that this is caching code, but maybe I'm wrong. I'll check it again and come back with a final answer.

Reply:
You're right! these tests failed because of the vpn on my mac. I will remove reshresh in this spec.